src/components/structures/RoomView.tsx

```diff
@@ -587,18 +540,18 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
             // The rest will be lost for now, until the aggregation API on the server
             // becomes available to fetch a whole thread
             if (!initialEvent) {
-                initialEvent = await fetchInitialEvent(this.context.client, roomId, initialEventId);
+                initialEvent = (await fetchInitialEvent(client, roomId!, initialEventId)) || undefined;
```

Comment:
`roomId` has to be nullish (as does `room`). It's a state variable, so it can't be known until the component mounts and the `RoomViewStore` is checked. Immediately after the component mounts, `roomId` is (basically) guaranteed to be defined, as long as the entire app is in a valid state. I agree that it's messy, but I think the scope of this PR is already an issue without more sweeping changes.

There is a test earlier in the method, it's just an ordering artifact I'll clear up.